src/patternfly/components/FormControl/form-control.scss

```
@@ -263,196 +187,177 @@ $pf-v5-c-form-control--m-clock--Coordinates: "M256 8C119 8 8 119 8 256s111 248 2
     border-bottom-width: var(--#{$form-control}--m-expanded--BorderBottomWidth);
   }
 
-  &:disabled {
+  &.pf-m-disabled {
     --#{$form-control}--BackgroundColor: var(--#{$form-control}--disabled--BackgroundColor);
+    --#{$form-control}--Color: var(--#{$form-control}--disabled--Color);
+    --#{$form-control}__toggle-icon--Color: var(--#{$form-control}--disabled__toggle-icon--Color);
 
-    color: var(--#{$form-control}--disabled--Color);
     cursor: not-allowed;
     border-color: var(--#{$form-control}--disabled--BorderColor);
   }
 
-  &[aria-invalid="true"] {
-    --#{$form-control}--PaddingRight: var(--#{$form-control}--invalid--PaddingRight);
-    --#{$form-control}--BorderBottomColor: var(--#{$form-control}--invalid--BorderBottomColor);
+  &.pf-m-error {
+    --#{$form-control}--PaddingRight: var(--#{$form-control}--m-error--PaddingRight);
+    --#{$form-control}--BorderBottomColor: var(--#{$form-control}--m-error--BorderBottomColor);
+    --#{$form-control}__icon--m-status--Color: var(--#{$form-control}--m-error__icon--Color);
+    --#{$form-control}__icon--Height: calc(var(--#{$form-control}--Height) - var(--#{$form-control}--m-error--BorderBottomWidth));
+    --#{$form-control}__toggle-icon--Height: calc(var(--#{$form-control}--Height) - var(--#{$form-control}--m-error--BorderBottomWidth));
+    --#{$form-control}--PaddingBottom: var(--#{$form-control}--m-error--PaddingBottom);
 
-    padding-bottom: var(--#{$form-control}--invalid--PaddingBottom); // Can't redefine --#{$form-control}--PaddingBottom b/c the hover padding bottom uses the focus padding var to compute it's padding
-    background-image: var(--#{$form-control}--invalid--BackgroundUrl);
-    background-position: var(--#{$form-control}--invalid--BackgroundPosition);
-    background-size: var(--#{$form-control}--invalid--BackgroundSize);
-    border-bottom-width: var(--#{$form-control}--invalid--BorderBottomWidth);
+    border-bottom-width: var(--#{$form-control}--m-error--BorderBottomWidth);
 
     &.pf-m-icon {
       --#{$form-control}--PaddingRight: var(--#{$form-control}--m-icon--icon--PaddingRight);
-
-      background-image: var(--#{$form-control}--m-icon--invalid--BackgroundUrl);
-      background-position: var(--#{$form-control}--m-icon--invalid--BackgroundPosition);
-      background-size: var(--#{$form-control}--m-icon--invalid--BackgroundSize);
     }
   }
 
   &.pf-m-success {
-    --#{$form-control}--PaddingRight: var(--#{$form-control}--success--PaddingRight);
-    --#{$form-control}--BorderBottomColor: var(--#{$form-control}--success--BorderBottomColor);
+    --#{$form-control}--PaddingRight: var(--#{$form-control}--m-success--PaddingRight);
+    --#{$form-control}--PaddingBottom: var(--#{$form-control}--m-success--PaddingBottom);
+    --#{$form-control}--BorderBottomColor: var(--#{$form-control}--m-success--BorderBottomColor);
+    --#{$form-control}__icon--m-status--Color: var(--#{$form-control}--m-success__icon--Color);
```

This should be updated for the other states, too

```suggestion
    --#{$form-control}__icon--m-status--Color: var(--#{$form-control}--m-success__icon--m-status--Color);
```